DataFormats/PatCandidates/interface/Photon.h

```diff
@@ -169,6 +170,11 @@ namespace pat {
 	    //<< key << " was not stored for this particle.";
           }
       }
+
+      float puppiChargedHadronIso() const {return puppiChargedHadronIso_; };
+      float puppiNeutralHadronIso() const {return puppiNeutralHadronIso_; };
+      float puppiPhotonIso() const {return puppiPhotonIso_; };
+
```

Comment:
The idea was to define puppi *without* candidates for which isolation is being calculated. Thus for electrons it is "NoLeptons", then for photons one has to define "NoPhotons" which is not defined clearly and we never studied that. Anyway, implementing this sort of thing is beyond this PR.